GenericsHomework/Node.cs

```diff
@@ -0,0 +1,120 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace GenericsHomework;
+public class Node<T>
+{
+    public T Value { get; }
+    public Node<T> Next { get; private set; }
+
+    public Node(T value)
+    {
+        Value = value;
+        Next = this;
+    }
+
+    public void SetNext(Node<T> next)
+    {
+        // I think having the check has made it so that next cannot be null
+        ArgumentNullException.ThrowIfNull(next);
+        this.Next = next;
+    }
+
+    public override string ToString()
+    {
+        Node<T> dummy = new(this.Value)
+        {
+            Next = this
+        };
+        string ret = "";
+        // goes through linked list starting at node called
+        while (dummy.Next.Next != this)
+        {
+            if (dummy.Next.Value != null)
+            {
+
+                ret += dummy.Next.Value.ToString() + " ";
+            }
+            dummy.Next = dummy.Next.Next;
+
+        }
+        if (dummy.Next.Value != null)
+        {
+
+            ret += dummy.Next.Value.ToString();
+        }
+
+        return ret;
+    }
+    public void Append(T value)
+    {
+        ArgumentNullException.ThrowIfNull(value);
+        if (Exists(value))
+        {
+            return;
+        }
+        Node<T> dummy = new(value)
+        {
+            Next = this
+        };
+        while (dummy.Next.Next != this)
+        {
+            dummy.Next = dummy.Next.Next;
+        }
+
+        Node<T> newNode = new(value);
+        dummy.Next.Next = newNode;
+        newNode.Next = this;
```

Comment:
```suggestion
     if (Exists(value)) throw new ArgumentException("This item already exists in the list");
     Node<T> newNode = new(value)
     {
         Next = Next
     };
     Next = newNode;
     return newNode;
```
My suggested implementation